lib/jnpr/junos/device.py

```diff
@@ -550,7 +553,14 @@ def execute(self, rpc_cmd, **kvargs):
         """
 
         if self.connected is not True:
-            raise EzErrors.ConnectClosedError(self)
+            if self.autoreconnect > 0:
+                self.autoreconnect -= 1
+                self.open()
```

Comment:
If the reconnect fails, this call will raise one of the exceptions from `open()`. These exceptions are not caught, effectively making any value for `autoreconnect > 1` useless. In addition, the caller of `execute()` is not expecting these new exception types to be raised.

Instead, I believe lines 557 and 558 should be changed to something like:
```
try:
    self.open()
# All exceptions returned by open() are a subclass of ConnectError
except EzErrors.ConnectError:
    self.autoreconnect -= 1
```